src/platform/mod.rs

```diff
@@ -32,12 +32,20 @@ pub mod windows;
 #[cfg(x11_platform)]
 pub mod x11;
 
+#[cfg(any(
+    windows_platform,
+    macos_platform,
+    x11_platform,
+    wayland_platform,
+    android_platform
+))]
+pub mod run_ondemand;
+
 #[cfg(any(
     windows_platform,
     macos_platform,
     android_platform,
     x11_platform,
     wayland_platform,
-    orbital_platform
 ))]
-pub mod run_return;
+pub mod pump_events;
```

Comment:
As you said, this is non-performant on macOS, but it's also quite bug-prone. Like sure, an application using it will "work", for some definition of work, but it will not really work _properly_: When resizing, the event loop will not yield back to the user/developer, in the end leading to an application that feels very janky.

So ideally, I would like to avoid having this implemented for macOS, since I would never want anyone to use it.

`run_ondemand` sounds more reasonable to my ears, though I suspect that even then the better thing to do would be to recommend users to start a new process instead. As an example, try adding `std::thread::sleep_ms(10000);` between the two invocations; you'll see that it looks like the application is unresponsive, even though it might actually be doing something in background.

Reply:
Yeah, those are all good points, and I would of course prefer an application that supports macOS over one that doesn't, even if it's janky.

And yeah, good documentation about the caveats (as well as maybe a note/policy that issues with it will have a lower priority) will probably mitigate most of my concerns.